memgpt/agent.py

```diff
@@ -916,13 +977,27 @@ def remove_function(self, function_name: str) -> str:
     #        self.ms.update_agent(agent=new_agent_state)
 
     def update_state(self) -> AgentState:
-        updated_state = {
-            "persona": self.memory.persona,
-            "human": self.memory.human,
-            "system": self.system,
-            "functions": self.functions,
-            "messages": [str(msg.id) for msg in self._messages],
-        }
+        if isinstance(self.memory, InContextMemory):
+            updated_state = {
+                "persona": self.memory.persona,
+                "human": self.memory.human,
+                "core_memory_type": "default",
+                "system": self.system,
+                "system_template": self.system_template,
+                "functions": self.functions,
+                "messages": [str(msg.id) for msg in self._messages],
+            }
+        elif isinstance(self.memory, CustomizableCoreMemory):

```

Comment:
Maybe a better name for `CustomizableCoreMemory` is `CustomizableInContextMemory`?

Reply:
It is now named in the same way as CoreMemory(CustomizableInContextMemory in agent.py).